var/spack/repos/builtin/packages/silo/package.py

```diff
@@ -103,6 +106,10 @@ def repl(match):
 
         filter_file(r'\b(DOMAIN|RANGE|UNION)\b', repl, *files_to_filter)
 
+    # Update autoconf's tests whether libtool supports shared libraries.
+    # (Otherwise, shared libraries are always disabled.)
+    force_autoreconf = True
```

Comment:
Ideally this would only be true when `+shared`, and (if/when a new version of silo is released). Try this:
```
    @property
    def force_autoreconf(self):
        # Update autoconf's tests whether libtool supports shared libraries.
        # (Otherwise, shared libraries are always disabled.)
        return self.spec.satisfies('+shared')
```
and then add `, when='+shared'` to the autotools deps.

Reply:
The shipped `configure` is now 7 years old; I thought that updating it would be a good idea anyway. What do you think?

I also realize now that "always disabled" should probably be "always disabled on Darwin"; I'll update the comment.

Reply:
Sorry, missed this among the other stuff in my notifications box 😅 you're probably right.  Next time someone hits a silo configure  bug let's just do that with a note about how old it is.